lib/fluent/plugin/out_secondary_file.rb

```diff
@@ -0,0 +1,162 @@
+#
+# Fluentd
+#
+#    Licensed under the Apache License, Version 2.0 (the "License");
+#    you may not use this file except in compliance with the License.
+#    You may obtain a copy of the License at
+#
+#        http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS,
+#    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#    See the License for the specific language governing permissions and
+#    limitations under the License.
+#
+
+require "fileutils"
+require "fluent/plugin/file_util"
+require "fluent/plugin/output"
+require "fluent/config/error"
+
+module Fluent::Plugin
+  class SecondaryFileOutput < Output
+    Fluent::Plugin.register_output("secondary_file", self)
+
+    SUPPORTED_COMPRESS = {
+      "gz" => :gz,
+      "gzip" => :gz,
+    }
+
+    FILE_PERMISSION = 0644
+    DIR_PERMISSION = 0755
+
+    desc "The Path of the file."
+    config_param :path, :string
+    desc "The flushed chunk is appended to existence file or not."
+    config_param :append, :bool, default: false
+    config_param :compress, default: nil do |val|
+      SUPPORTED_COMPRESS[val].tap do |e|
+        raise ConfigError, "Unsupported compression algorithm '#{val}'" unless e
+      end
+    end
+
+    def initialize
+      super
+    end
+
+    def configure(conf)
+      unless @as_secondary
+        raise Fluent::ConfigError, "This plugin can only be used in the <secondary> section"
+      end
+
+      unless @path = conf["path"]
```

Comment:
Thanks. I don't know that.
Fixed.
https://github.com/fluent/fluentd/pull/1154/files#diff-68f04670661c50196427de0d60c06636R35